exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetIsPredicate.java

```diff
@@ -114,8 +115,7 @@ public boolean canDrop(RangeExprEvaluator<C> evaluator) {
    */
   private static LogicalExpression createIsTruePredicate(LogicalExpression expr) {
     return new ParquetIsPredicate<Boolean>(expr, (exprStat, evaluator) ->
-        //if max value is not true or if there are all nulls  -> canDrop
-        isAllNulls(exprStat, evaluator.getRowCount()) || exprStat.hasNonNullValue() && !((BooleanStatistics) exprStat).getMax()
+      exprStat.hasNonNullValue() && !((BooleanStatistics) exprStat).getMax() || isAllNulls(exprStat, evaluator.getRowCount()) ? RowsMatch.NONE : checkNull(exprStat)
```

Comment:
Please make sure to keep the original order of expressions unless it needs to be changed for a reason. If there is a reason to change it here or *in any other places*, please provide an explanation. I don't see why it needs to be changed.

Reply:
OK. Reversed.